### ISSUE TITLE
Siblings of music files without an Album tag get all grouped together in one MusicAlbum

### ISSUE THREAD

**Issue by Aanok** [jellyfin/jellyfin#2879]
**Describe the bug**
Consider the following folder under a music library:
```
fabrizio@homeserver:/mnt/storage/test_music/The Oscillation$ ls
2010 - Future Echo e.p                              2014 - Travel Expop 2 Great Britain  2018 - Wasted Space
2011 - Dead Cell Memory                             2014 - Waste The Day                 2020 - Droneweapon
2011 - Fall e.p                                     2015 - Beyond The Mirror             2020 - The Oscillation live at Audio Underground 2011
2011 - Veils                                        2015 - New Way To Feel               From Tomorrow (Blue Sun ambient mix).opus
2013 - From Tomorrow                                2016 - Monographic                   Subconsiousness (Future Echo drum machine early version).opus
2014 - Cable Street Sessions                        2017 - Evil In The Tree              Visitation Exit (early version 2006).opus
```
where subfolders correspond to albums, files inside the subfolders are tagged with Artist, Title and Album but those last three loose files have Artist and Title but not an Album metadata tag.

The presence of those last three files causes Jellyfin to wrongly present **all** files as children of just one MusicAlbum, in this case `Live at Beursschourwburg, Brussels Jan 2014`.

Here's a [screenshot](https://user-images.githubusercontent.com/18417628/79325889-3dba9b80-7f12-11ea-9171-c17ab6635100.jpg).

[Here](https://github.com/jellyfin/jellyfin/files/4480652/oscillation.json.txt) is the JSON object returned by querying the API endpoint for `Live at Beursschourwburg, Brussels Jan 2014`.

Note how the Audio items returned by the query have the correct Album but not AlbumId fields when they correspond to items in a subfolder, while those for the three loose files are missing an Album field altogether.

**System:**
 - OS: Debian 9.12
 - Virtualization: No
 - Clients: All
 - Jellyfin Version: 10.5.4
 - Storage: local

**To Reproduce**
1. Organize files on disc as described, i.e. have a folder for an artist with subfolders for albums, all correctly tagged, as well as some loose files without Album tags under the artist parent folder.
2. Scan the files into a music library.

**Expected behavior**
What is important is that **the files correctly tagged and organized on the file system should be correctly recognized and presented by Jellyfin**. All those albums should show up, with their content, no matter what.

There is then some discussion to be had on what exactly to do with the loose files both in terms of database and presentation to the user.

The obvious approach would be to group them under some sort of fake album (`Unknown Album` is a popular choice in other software): this could be done manually by the user (by giving them their own subfolder or tagging them with an arbitrary Album) or better automatically by Jellyfin.

This would however require Jellyfin to then treat this fake album in a special way (which I imagine would complicate things on the database side), as the files within may differ in fields that are assumed uniform for a MusicAlbum, namely Year, and may present tags that have no meaning in the new context, like Track number or CD number. A not ideal workaround may be for the Jellyfin scanner to just drop this kind of information altogether.

**Logs**
The logs show no errors or warnings.

**Comment by emaen** [comment]
Hello !
It seems you have the same issue than me : 
[https://github.com/jellyfin/jellyfin/issues/2806](url)

I've got the same result.
I thought this was related to "AlbumArtist" tag tagged with "Various Artists" in all these files. But It may be related to single 'non-albums" files presence, as you said.

**Comment by Aanok** [comment]
You should double check if you have some files missing an Album tag, then. If so, then this issue is indeed a direct duplicate of #2806, otherwise they are probably just related.

I should note this problem manifests only when the loose files are inside a folder corresponding to an artist. I have lots of loose files in the root of my music library that are recognized as children of the library itself (speaking in terms of `parentid`) and pose no problem.

**Comment by emaen** [comment]
hello !
I've triple checked the files. One was with no album tag. I've deleted it.
Then I've cleaned the whole folder ( "Various Artists" ), verified all tracks from all albums, deleted .nfo files, deleted firefox cache, re-scanned the library.

Result is the same : all files are listed under one unique album !!

**Comment by gowon** [comment]
I can confirm the same issue as OP; any folder in my music library with loose tracks with no Album causes all the tracks in the root of that folder to be combined into a single album. Since my folder structure follows an Artist > Album heirarchy, I wind up with a lot of artists having their whole catalogs combined to one mega album.

**System:**

- OS: Synology DSM 6.2.3-25426
- Virtualization: Docker, jellyfin/jellyfin:latest
- Clients: All
- Jellyfin Version: 10.5.5
- Storage: local

**Comment by xcorvis** [comment]
I also have a group of singles all stuck into one album, but I don't believe it's due to the lack of an album tag. All of mine do have album tags. All include title, artist, albumartist and album at minimum.

It is possible that some files didn't originally have album tags, but I just tried removing the library, clearing cache and metadata and re-adding it, and it's the same. I have not gone so far as to completely wipe and reinstall jellyfin. I don't know if it's likely to hang on to that info through that process.

This may or may not be related bug - in my Admin->Metadata->Music menu, I do not see artists whose names start with numbers, such as "38 special" or "3 doors down", and I do have tracks from both of them in my mixed up album. I see "Motörhead" (non-ascii chars) in the metadata, and when I examine the individual tracks they do have the correct artist names and otherwise appear normal. 

I did some searching and found this issue, which has more info: #3169

**Comment by gowon** [comment]
I believe the issue is caused by file structure, not ID3 metadata. If in the tree of your Media folder, you have a folder that contains both music files and subfolders with music files in them, it will cause Jellyfin to catalog all of those files as if they are part of a single album.

My convention for organizing my music files is:

- Artist folder
  - Artist music without Album tags
  - Album folder
    - Artist music with Album tags

Any situation in which I have loose music and album folders for an artist, the artist's whole catalog was indexed into a single album (with the album name being the Artist folder name), regardless of tagging. The only way I've found to work around this bug is to put all music files in their own folder, so that in your Media folder you only have:

1. Folders that contain only music files
2. Folders that contain only subfolders
3. **No folders that contain a mix of music files and subfolders** 

Quite a hassle. Somewhere in the logic the file structure is overriding ID3 parsing during the scanning and indexing process.

**Comment by cvium** [comment]
Should be fixed by #4968
